core/io/resource_format_binary.cpp

```diff
@@ -2244,12 +2244,33 @@ Error ResourceFormatSaverBinaryInstance::save(const String &p_path, const Ref<Re
 		if (r->is_built_in()) {
 			if (r->get_scene_unique_id().is_empty()) {
 				String new_id;
-
+				int attempt = 0;
 				while (true) {
-					new_id = _resource_get_class(r) + "_" + Resource::generate_scene_unique_id();
+					String long_name = _resource_get_class(r) + "_" + r->get_name() + "_" + itos(res_index);
+					if (attempt != 0) {
+						long_name += "_" + itos(attempt);
+					}
+
+					uint32_t hash = long_name.hash();
+
+					// Compact hash as in `Resource::generate_scene_unique_id()`.
+					static constexpr uint32_t characters = 5;
+					static constexpr uint32_t char_count = ('z' - 'a');
+					static constexpr uint32_t base = char_count + ('9' - '0');
```

Comment:
```suggestion
					static constexpr uint32_t char_count = ('z' - 'a') + 1;
					static constexpr uint32_t base = char_count + ('9' - '0') + 1;
```
Given:
* https://github.com/godotengine/godot/issues/83843

Reply:
Ah yes I saw that interesting issue. I should find a way to reuse the function than write it twice.

Reply:
The one for UIDs unfortunately can't change right now because of compatibility, but when making new code with it it'd be good to get it correct, the UID code now has a note next to it as well to clarify this

Adding a method to encode it in one place would be best indeed!